replce windows_batch with batch in import_certificate.rb

Es-Paul opened the pull request:
Hi Matt,

The windows_batch resource has been removed from the windows cookbook, since newer versions of chef 
have a "batch" resource.

This has caused the import_certificate.rb provider to fail with newer versions of the windows cookbook.
I have updated the import_certificate.rb provider to use the new "batch" provider.

Please accept my pull request.

Cheers,
Paul

Es-Paul committed:
replce windows_batch with batch in import_certificate.rb

the windows_batch resource has been removed from the windows cookbook, since newer versions of chef have a "batch" resource

nealbrown commented:
I was about to open the same PR

I replaced windows_batch with execute and code with command, same reason

Es-Paul committed:
Changed the 'hash' attribute to 'certificate_hash'

    naming an attribute 'hash' is incompatible with chef-client 13

Updated supported OS list

    We have tested this on Server 2012 and Server 2016 with no issues.
    Server 2008 R2 also works for now.
    Server 2008 may no longer be compatible
